plugins/codemirror/plugin.js

```diff
@@ -0,0 +1,130 @@
+(function() {
+	'use strict';
+
+	var stylesLoaded = false;
+
+
+	CKEDITOR.plugins.add('codemirror', {
+		_createCodeMirrorEditor: function(editor) {
+			var instance = this;
+
+			var codemirror = CKEDITOR.plugins.codemirror;
+
+			editor.addMode('source', function(callback) {
+				var contentsSpace = editor.ui.space('contents');
+
+				var textarea = contentsSpace.getDocument().createElement('textarea');
+				contentsSpace.append(textarea);
+
+				instance.codeMirrorEditor = CodeMirror.fromTextArea(
+					textarea.$,
+					{
+						lineNumbers: true,
+						mode: 'text/html'
+					}
+				);
+
+				var oldData = editor.getData(1);
+
+				instance.codeMirrorEditor.setValue(html_beautify(oldData));
+
+				instance.codeMirrorEditor.on('change', instance._handleCodeMirrorChange.bind(instance, editor, oldData));
+
+				editor.on('resize', instance._handleEditorResize.bind(instance));
+
+				editor.fire('ariaWidget', this);
+
+				callback();
+			});
+
+			editor.addCommand('codemirror', codemirror.commands.source);
```

Comment:
@carloslancha this is my fault and I was going to change it in my previous PR:
I'm pretty sure this should go on line 100 (or at least in the `init` function). 

I'm not getting any error in dxp, but if I test our build of `ckeditor` in simple html page I'm getting this error because the `codemirror` command isn't defined in the `mode` event listener (declared on line 101)

![error-01](https://user-images.githubusercontent.com/5572/91142843-d081f880-e6b1-11ea-9478-9ea2166e65a5.png)

Coud you please have a look? 
Thanks

Reply:
Mmmm, shouldn't we move
```
if (editor.ui.addButton) {
				editor.ui.addButton('Source', {
					label: editor.lang.codemirror.source,
					command: 'codemirror',
					toolbar: 'mode,10'
				});
			}

			editor.on('mode', function() {
				editor.getCommand('codemirror').setState(
					editor.mode === 'source'
					? CKEDITOR.TRISTATE_ON
					: CKEDITOR.TRISTATE_OFF
				);
			});
``` 

to the end of `_createCodeMirrorEditor` so we're sure that everything is loaded and initialized?

Reply:
Yeah, that's actually a good idea.
Let me try that too though, just to be sure :+1:

Reply:
@carloslancha after looking at this, we should keep this in the `init` function, because if we call it in the `_createCodeMirrorEditor` function the "source" button won't be available in the toolbar as you can see in this screenshot 

![](https://user-images.githubusercontent.com/5572/91149178-0b3c5e80-e6bb-11ea-90dc-72c30022eb10.png)